### PR TITLE
Update build.hpp

### PR DESCRIPTION
**Pull request by Anmol2001**
I have documented the updates with the test commands and also added a memory warning with the make -j N command.

### CODE REVIEW

**Review by zoq — APPROVED** [review]
Looks good to me, thanks.

**Review by gauravghati — APPROVED** [review]
Looks good for me too.

File: doc/guide/build.hpp [hunk 1]
```diff
@@ -249,11 +250,37 @@ suite.  You can build this component with
 $ make mlpack_test
 @endcode
 
-and then run all of the tests, or an individual test suite:
+We use Catch2 to write our tests
```
Comment (rcurtin) [inline doc/guide/build.hpp:253]:
```suggestion
We use <a href="https://github.com/catchorg/Catch2">Catch2</a> to write our tests.
```
Might be nice to include a URL too. :)

File: doc/guide/build.hpp [hunk 2]
```diff
@@ -249,11 +250,37 @@ suite.  You can build this component with
...
+
+@code
+$ ./bin/mlpack_test "[testname]"
+@endcode
+
+where testname is the name of the test suite. 
+For example to run all collaborative filtering tests implemented in cv_test.cpp you can run:
+
+@code
+./bin/mlpack_test "[CVTest]"
```
Comment (rcurtin) [inline doc/guide/build.hpp:270]:
```suggestion
For example to run all collaborative filtering tests implemented in cf_test.cpp you can run:

@code
./bin/mlpack_test "[CFTest]"
```

File: doc/guide/build.hpp [hunk 3]
```diff
@@ -249,11 +250,37 @@ suite.  You can build this component with
...
+@endcode
+
+Now similarly you can run all the binding related tests using:
+
+@code
+./bin/mlpack_test "[BindingTests]"
+@endcode
+
+To run a single test, you can explicitly provide the name of the test, for example,
```
Comment (rcurtin) [inline doc/guide/build.hpp:279]:
```suggestion
To run a single test, you can explicitly provide the name of the test; for example, 
```
Just a pedantic grammar comment. :)

**Review by rcurtin — APPROVED** [review]
@Anmol2001 nice!  Thanks for improving our documentation.  I have a couple tiny little comments that might be worth including. :+1:

File: doc/guide/build.hpp (hunk 1 above)
Comment (Anmol2001) [reply doc/guide/build.hpp:253]: @rcurtin Yes sure ..... I think i will be good to use this same URL for Catch2 here in test docs readme.md(https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/README.md) as the existing one is showing 404 error.

**Comment by Anmol2001** [comment]
> @Anmol2001 nice! Thanks for improving our documentation. I have a couple tiny little comments that might be worth including. 👍
Thanks @rcurtin  for the nice suggestions !  I have committed the suggestions as told by you and did one  change of link(for Catch 2) in the test doc Readme file  -> dd442ac6b15e981fc1778ad36efa6d1b2edcbd3d .

File: src/mlpack/tests/README.md [hunk 4]
```diff
@@ -43,4 +43,4 @@ To run a single test, you can explicitly provide the name of the test, for examp
 
 `./bin/mlpack_test BinaryClassificationMetricsTest`
 
-Catch2 provides many other features like filter, checkout the [Catch2 reference section](docs/Readme.md#top) - for more details.
+Catch2 provides many other features like filter, checkout the [Catch2 reference section](https://github.com/catchorg) - for more details.
```
Comment (rcurtin) [inline src/mlpack/tests/README.md:46]: Should we link to here instead? https://github.com/catchorg/Catch2/blob/devel/docs/Readme.md#top

Comment (Anmol2001) [reply src/mlpack/tests/README.md:46]: yes that is more convenient .. i am changing : )

File: doc/guide/build.hpp (hunk 1 above)
Comment (rcurtin) [reply doc/guide/build.hpp:253]: Sounds good---do you want to include the link here in `build.hpp` too?

**Review by shrit — APPROVED** [review]

**Pull request merged**